test_handler.go

```diff
@@ -110,3 +124,37 @@ func GetBrowserNames() (browserNames []string, err error) {
 	sort.Strings(browserNames)
 	return browserNames, nil
 }
+
+// getLastCompleteRunSHA returns the SHA[0:10] for the most recent run that complete for all of the given browser names.
+func getLastCompleteRunSHA(ctx context.Context, browserNames []string) (sha string, err error) {
+	baseQuery := datastore.
+		NewQuery("TestRun").
+		Order("-CreatedAt").
+		Limit(100).
```

Comment:
I believe the results of this query are memcache'd (they would be in Python/ndb, not sure about Go), but we should double check the performance hit of this. I can do a quick check when deploying.